l10n_nl_xaf_auditfile_export/models/xaf_auditfile_export.py

```diff
@@ -179,21 +178,9 @@ def get_odoo_version(self):
         return release.version
 
     def get_partners(self):
-        """return a generator over partners"""
-        partner_ids = (
-            self.env["res.partner"]
-            .search(
-                [
-                    "|",
-                    ("customer_rank", ">", 0),
-                    ("supplier_rank", ">", 0),
-                    "|",
-                    ("company_id", "=", False),
-                    ("company_id", "=", self.company_id.id),
-                ]
-            )
-            .ids
-        )
+        """return a generator over relevant partners"""
+        relevant_partners=self.env["account.move.line"].search([("company_id","=",self.company_id.id),("date",">=",self.date_start),("date","<=",self.date_end)]).partner_id
```

Comment:
Will this not be very slow for databases with a huge amount of transactions per period?

SELECT id FROM res_partner WHERE id IN (SELECT partner_id FROM account_move_line
 WHERE company_id = %(company_id)s AND date >= %(date_start)s AND date <= %(date_end)s"

Maybe better use SQL to get the partner ids and then browse those?

Reply:
I agree with @NL66278 here - there are systems with loads and loads of account move line records, and the ORM statement you wrote will either load them all in memory and then extract the unique partner_ids via loop (= detrimental to memory usage) or come up with the above nested SQL, which would be slow because nested SQL queries do not make use of indexes effectively.

Since it's just a small change I'd go for this one just to be on the safe side:

```
self.env.cr.execute(
    "select distinct partner_id "
    "from account_move_line "
    "where company_id = %s "
    "and date >= %s "
    "and date <= %s ",
    (self.company_id.id, self.date_start, self.date_end),
)
partners = self.env['res.partner'].browse([pid for pid in self.env.cr.fetchall()])  # perhaps with sudo
```